integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java

```diff
@@ -132,6 +133,27 @@ public void testRefreshAfterMergeWithIcebergTableCaching()
         .containsExactlyInAnyOrder(row(23, "test"), row(24, "test24"));
   }
 
+  @Test
+  public void testRenameTableToMissingNamespace()
+    throws NessieNotFoundException, NessieNamespaceAlreadyExistsException {
+    createBranchForTest(refName);
+    api.createNamespace().refName(refName).namespace(Namespace.of("db")).create();
+    Reference ref = api.getReference().refName(refName).get();
+
+    assertThat(sql("USE REFERENCE %s IN nessie", refName))
+      .hasSize(1)
+      .containsExactly(row("Branch", refName, ref.getHash()));
+
+    sql("CREATE TABLE nessie.db.tbl (id int, name string)");
+    sql("INSERT INTO nessie.db.tbl select 23, \"test\"");
+    assertThat(sql("SELECT * FROM nessie.db.tbl")).hasSize(1).containsExactly(row(23, "test"));
+
+    // rename to the table identifier with non-existing namespace
+    assertThatThrownBy(() -> sql("ALTER TABLE nessie.db.tbl RENAME TO nessie.non_existing.table"))
+      .isInstanceOf(RuntimeException.class)
+      .hasMessage("Cannot rename table 'tbl' to 'table': the current reference is not up to date");
```

Comment:
This final error from iceberg client is `CommitFailedException`, because we are using `NessieUtil.maybeThrowSpecializedException` only for commit operations. We need to fix it at the Iceberg side.

```
org.apache.iceberg.exceptions.CommitFailedException: Cannot rename table 'tbl' to 'table': the current reference is not up to date.
	at org.apache.iceberg.nessie.NessieIcebergClient.renameTable(NessieIcebergClient.java:366)
	at org.apache.iceberg.nessie.NessieCatalog.renameTable(NessieCatalog.java:283)
	at org.apache.iceberg.CachingCatalog.renameTable(CachingCatalog.java:178)
	at org.apache.iceberg.spark.SparkCatalog.renameTable(SparkCatalog.java:389)
	at org.apache.spark.sql.execution.datasources.v2.RenameTableExec.run(RenameTableExec.scala:51)
```

Actual server side error
```
Caused by: org.projectnessie.error.NessieReferenceConflictException: Namespace 'non_existing' must exist.
	at org.projectnessie.error.ErrorCode.lambda$asException$1(ErrorCode.java:66)
```